core/src/text/fontContext.cpp

```diff
@@ -40,35 +45,54 @@ void FontContext::unlock() {
     m_contextMutex.unlock();
 }
 
-bool FontContext::addFont(const std::string& _family, const std::string& _weight, const std::string& _style) {
+FontID FontContext::addFont(const std::string& _family, const std::string& _weight,
+                            const std::string& _style, bool _tryFallback) {
+
+    unsigned int dataSize = 0;
+    unsigned char* data = nullptr;
+    int font = FONS_INVALID;
 
     std::string fontKey = _family + "_" + _weight + "_" + _style;
-    if (m_fonts.find(fontKey) != m_fonts.end()) {
-        return true;
-    }
 
-    unsigned int dataSize;
-    unsigned char* data = nullptr;
+    auto it = m_fonts.find(fontKey);
+    if (it != m_fonts.end()) {
+        if (it->second < 0) {
+            goto fallback;
```

Comment:
Not gone through the code yet to give any opinion right now on usage of `goto` :D.

But I agree exceptions are in the same league of goto statements.